rust/kernel/types.rs

```diff
@@ -91,3 +91,33 @@ impl<T: PointerWrapper + Deref> PointerWrapper for Pin<T> {
         Pin::new_unchecked(T::from_pointer(p))
     }
 }
+
+/// Runs a cleanup function/closure when dropped.
+///
+/// The [`ScopeGuard::dismiss`] function prevents the cleanup function from running.
```

Comment:
Please add two or three examples, e.g. a trivial one with a explicit scope for extra clarity, one with `dismiss()` to show the difference and one with the `?` operator etc. in the middle like in your code.

Reply:
Added two examples.